File: yadirect-strapi/src/cron-tasks/yandex-api/AccountBalance.ts
export default class AccountBalance {
  accountId: string;
  agencyName: string;
  amount: number;
  currency: string;
  login: string;

  constructor(accountId: string, agencyName: string, amount: number, currency: string, login: string) {
    this.accountId = accountId;
    this.agencyName = agencyName;
    this.amount = amount;
    this.currency = currency;
    this.login = login;
  }

  static build(payload: any): AccountBalance | null {
    try {
      return new AccountBalance(
        payload['AccountID'],
        payload['AgencyName'],
        parseFloat(payload['Amount']),
        payload['Currency'],
        payload['Login']
      );
    } catch (e) {
      console.error('An error occurred while parsing account balance: ' + e);
      return null; // Return null in case of an error
    }
  }
}

File: yadirect-old/yandexDirectBot/src/yandex_direct_api/account_balance.py
import logging


class AccountBalance:
    account_id: str
    agency_name: str
    amount: int
    currency: str
    login: str

    @staticmethod
    def build(payload: dict):
        try:
            account_total = AccountBalance()

            account_total.account_id = payload['AccountID']
            account_total.agency_name = payload['AgencyName']
            account_total.amount = float(payload['Amount'])
            account_total.currency = payload['Currency']
            account_total.account_id = payload['AccountID']
            account_total.login = payload['Login']

            return account_total
        except Exception as e:
            logging.error('An error occurred while parsing campaign:' + str(e))
